apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -1088,6 +1111,110 @@ defmodule Champions.Battle.Simulator do
     {target, history}
   end
 
+  defp process_executions_over_time(unit_initial_state, current_unit, history) do
+    Enum.reduce(unit_initial_state.executions_over_time, {current_unit, history}, fn execution_over_time,
+                                                                                     {unit_acc, history_acc} ->
+      process_execution_over_time(execution_over_time, unit_acc, history_acc, unit_initial_state)
+    end)
+  end
+
+  defp process_execution_over_time(
+         %{remaining_duration: -1} = execution_over_time,
+         target,
+         history,
+         _target_initial_state
+       ) do
+    # If the execution is over, we remove it from the target
+    new_target =
+      update_in(target, [:executions_over_time], fn current_executions ->
+        Enum.filter(current_executions, fn exec -> exec.skill_id != execution_over_time.skill_id end)
```

Comment:
What will happen if two characters with the same skill apply an effect to the same character in different times, isn't this going to cause that effect to be removed when the first one expires?

Reply:
Yes, that will turn into a problem. I changed it, same as I did in [fdafe26](https://github.com/lambdaclass/mirra_backend/pull/662/commits/fdafe26fc1dca556d7e9502c0a8cfd6e2396a69e)